src/core/BufferAttribute.js

```diff
@@ -310,6 +311,14 @@ BufferAttribute.prototype = {
 
 		return new this.constructor().copy( this );
 
+	},
+
+	discard: function () {
+
+		var oldArray = this.array;
+
+		this.array = new oldArray.constructor( 1 ); // create dummy minimal length TypedArray
```

Comment:
I've looked at the issue of keeping the minimal typed array again, and tried moving the 

"if ( array instanceof * ) then  type = gl.{TYPE) " 
...
sequence  from the webGLRenderer to webGLObjects' internal createBuffer() function. In createBuffer the appropriate gl.(type) and bytesPerElement values are added to the existing attribute properties object.  This way the "if" statements only run once, and the webGLRenderer now has no need to access an attribute's .array property. 

The disposeArray() can now be just replaced by .array = null in the callback, or used in disposeArray() if you prefer to hide the mechanism,  

How does this sound?